pkg/apis/rollouts/validation/validation.go

```diff
@@ -130,6 +130,12 @@ func ValidateRolloutSpec(rollout *v1alpha1.Rollout, fldPath *field.Path) field.E
 
 	allErrs = append(allErrs, ValidateRolloutStrategy(rollout, fldPath.Child("strategy"))...)
 
+	// WorkloadRef and template can not be set at the same time
+	if spec.WorkloadRef != nil && !spec.EmptyTemplate() {
+		allErrs = append(allErrs, field.InternalError(fldPath.Child("template"),
+			fmt.Errorf("template must be empty for workload reference rollout")))
+	}
+
```

Comment:
@jessesuen , I moved the check to validation. Could you take another look?

BTW, when I run the lint command on your example, I got

```
Error: spec.template.spec.containers: Required value
```

So the lint sub command works without the PR.

Reply:
After further investigation, I found that https://github.com/argoproj/argo-rollouts/blob/04201fb068dab917218e0533231f8007d2cba71c/pkg/apis/rollouts/validation/validation.go#L117

returns an error for empty template.spec.containers if using the lint sub-command. However, if we apply the invalid spec, the spec.containers are filled as default value at some where, which causes the issue

Reply:
so I think the fix could be move the validation to somewhere before controller fills out the template.spec.containers field so that the error can be caught earlier. what do you think?

Reply:
My above thoughts kind of complicate the problem. Please see if the updated commit looks good to you. Thanks.